src/Chess.java

```diff
@@ -13,45 +33,112 @@ public static void play() {
-                System.out.print("try again: ");
-                move = Move.parse(input.next());
-                if (move != null) piece = chessBoard.getPiece(move.src);
+            if (!saved) {
+                Move move = Move.parse(moveString);
+                if (move != null) {
+                    piece = this.chessBoard.getPiece(move.src);
+                }
+
+                while (move == null || this.chessBoard.moveIsInCheck(move)) {
+                    System.out.println("error: invalid move");
+                    System.out.print("try again: ");
+                    move = Move.parse(input.next());
+                    if (move != null) piece = this.chessBoard.getPiece(move.src);
+                }
+
+                this.chessBoard.makeMove(move);
+
+                System.out.println();
+
+                this.checkmate = this.chessBoard.isCheckmate(this.chessBoard.getCurrentPlayer());
+                if (!this.checkmate) this.stalemate = this.chessBoard.isStalemate(this.chessBoard.getCurrentPlayer());
+            
+                System.out.println();
+
+                if (this.checkmate) System.out.println("Game Over! " + this.chessBoard.getCurrentPlayer().opposite() + " Wins!");
+                if (this.stalemate) System.out.println("Game Over! It's a draw!");
             }
+        }
+    }
 
-            chessBoard.makeMove(move);
+    private Boolean saveGame(String fileName) {
 
-            System.out.println();
+        String board = "";
+        String turn = this.chessBoard.getCurrentPlayer().toString();
 
-            checkmate = chessBoard.isCheckmate(chessBoard.getCurrentPlayer());
-            if (!checkmate) stalemate = chessBoard.isStalemate(chessBoard.getCurrentPlayer());
+        for (int i = 0; i < Board.SIZE; i++) {
+            for (int j = 0; j < Board.SIZE; j++) {
+                Square sq = new Square(j, i);
+                Piece piece = this.chessBoard.getPiece(sq);
+                if (piece != null) {
+                    board += piece.kind;
+                    board += piece.color == Color.WHITE ? ",w" : ",b";
+                    board += " ";
+                } else {
+                    board += ". ";
+                }
+            }
         }
 
-        System.out.println(chessBoard);
-        System.out.println();
+        board = board.substring(0, board.length() - 1);
+
+        try {
+            File d = new File("ChessGames");
+            if (!d.isDirectory()) {
+                d.mkdir();
+            }
+            if (fileName.equals("")) {
+                fileName = "ChessGames/chess1";
+                File f = new File(fileName);
+                int i = 1;
+                while (f.exists() || f.isDirectory()) {
+                    fileName = fileName.substring(0, fileName.length() - 1);
+                    fileName += i;
+                    i++;
+                    f = new File(fileName);
+                }
```

Comment:
I think this is a slightly more readable way to do this:

```suggestion
                String prefix = "ChessGames/chess";
                File f = new File(fileName);
                int i = 1;
                do {
                    fileName = prefix + i;
                    i++;
                    f = new File(fileName);
                } while (f.exists() || f.isDirectory());
```